js/ui/drop_down_editor/ui.drop_down_list.js

```diff
@@ -326,6 +328,44 @@ var DropDownList = DropDownEditor.inherit({
         }
     },
 
+    _fitIntoRange: function(value, start, end) {
+        if(value > end) {
+            return start;
+        }
+        if(value < start) {
+            return end;
+        }
+        return value;
+    },
+
+    _items: function() {
+        var items = this._getPlainItems(!this._list && this._dataSource.items());
```

Comment:
Could we move a couple of tests from the SelectBox to test it (or add new tests for the DropDownList)

Reply:
This is the refactoring. This code was moved from the dxSelectBox to the dxDropDownList to make it possible to reuse it in the dxAutoComplete. The behavior was not changed. I don't know what tests is required